syntax/src/main/java/org/aya/syntax/ref/GeneralizedVar.java

```diff
@@ -6,17 +6,27 @@
 import org.aya.util.error.SourceNode;
 import org.aya.util.error.SourcePos;
 import org.jetbrains.annotations.NotNull;
+import kala.collection.immutable.ImmutableSeq;
 
 public final class GeneralizedVar implements AnyVar, SourceNode {
   public final @NotNull String name;
   public final @NotNull SourcePos sourcePos;
   public Generalize owner;
+  private @NotNull ImmutableSeq<GeneralizedVar> dependencies = ImmutableSeq.empty();
```

Comment:
A `@Nullable` dependencies with a `null` initial value can be better because we shouldn't use them before resolving.